src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XDocumentType.cs

```diff
@@ -53,14 +53,10 @@ internal XDocumentType(XmlReader r)
         /// <summary>
         /// Gets or sets the internal subset for this Document Type Definition (DTD).
         /// </summary>
-        public string InternalSubset
+        public string? InternalSubset
         {
             get
             {
-                // TODO-NULLABLE: As per documentation, this should return string.Empty.
```

Comment:
Are the docs going to be updated?

Reply:
I couldn't find any doc saying anything like that. I think @Jozkee meant XmlReader.Value doc which this returned directly (but I changed it to fall back to string.Empty). I've asked him offline what doc he's referring to clarify and will wait with merging until that's clear

Reply:
https://docs.microsoft.com/dotnet/api/system.xml.xmldocumenttype.internalsubset?view=netcore-3.1#System_Xml_XmlDocumentType_InternalSubset

> The DTD internal subset on the DOCTYPE. If there is no DTD internal subset, **String.Empty is returned**.

Reply:
that's XmlDocumentType not XDocumentType but I've tested some behavior to see which one is most consistent and I think it makes sense to accept null but return string.Empty:
- currently constructor and setter accept null
- when you load the xml without internal subset it returns string.Empty
- XmlDocumentType also returns string.Empty

so I'll update the property to be non-null but accepting null in the setter.

Reply:
> that's XmlDocumentType not XDocumentType

Oh I misread it, sorry about that.

Is **Xml**DocumentType.InternalSubset honoring the docs, though? If so, shoudn't that property have the same nullability characteristics as this one?